# 06. 두 모평균의 비교 및 두 범주형 변수 관계 파악

* 모집단의 평균값이 서로 다른지 확인하기 위한 방법: t-검정  

예) A지역과 B지역의(두 지역의 인구수는 같다고 가정) 평균연령을 조사하기 위하여 두 지역에서 각각 100명씩 조사하여 평균 연령이 41.2세와 40.5세로 나왔다면 두 지역의 전체 인구의 평균연령에는 차이가 있는지 확인하기 위해서는 t-검정을 사용할 수 있다. 각 지역별 100명의 평균연령은 각 개인의 나이와 같은 __연속형 변수__ 로 연속형 변수의 모집단 특성을 확인할 때 t-검정을 사용  

* 표본의 관측된 빈도수와 기대빈도의 차이를 바탕으로 두 집단의 동일성 과 독립성을 확인할 수 있는 방법: 카이제곱 검정  

예) A지역과 B지역의(두 지역의 인구수는 같다고 가정) 흡연율을 조사하기 위하여 두 지역에서 각각 100명씩 조사하여 평균 흡연률이 21%, 18%로 나왔다면 두 지역 전체 인구의 흡연률에는 차이가 있는지 확인하기 위해서는 카이제곱 검정을 사용할 수 있다. 개인의 흡연여부는 나이와 달리 흡연, 비흡연으로 나눠지는 __범주형 변수__ 이므로 범주형 변수의 모집단 특성을 확인할 때 카이제곱 검정을 사용  

> 연속형 변수 --> 모평균 차이 검정   

> 범주형 변수 --> 독립성 검정  

## 1. 독립표본 t-검정 (독립인 두 집단에 대한 모평균 차이비교)

### 1-1. 독립표본 t-검정
서로 독립인 __2개의 집단의 모평균에 유의미한 차이__ 가 있는지를 확인할 수 있는 통계적 검정 방법  

예) 일 평균 스마트폰 사용시간이 성별에 따라 다를까?, 동아리 활동 여부에 따라 학교생활 만족도에 차이가 있을까?  

1. 개념
독립인 두 정규모집단의 모평균에 차이가 있는지 검정
    - 각 모집단으로부터 추출된 독립표본을 이용할 수 있음
    - 두 정규모집단은 서로 __독립__ 이라 가정함  

예를 들어 서로 다른 종류의 목초를 먹인 소의 우유 생산량에 차이가 있는가? (귀무가설: u1=u2 / 대립가설: u1=/=u2)와 같은 문제를 비교하기 위함  


#### 정규성 검정: shaprio()
각 데이터셋의 정규성 확인을 위해 shapiro()함수를 적용

> shapiro(data)  


In [1]:
x1 = [44, 44, 56, 46, 47, 38, 58, 53, 49, 35, 46, 30, 41]
x2 = [35, 47, 55, 29, 40, 39, 32, 41, 42, 57, 51, 39]

In [2]:
from scipy.stats import levene, ttest_ind, shapiro

Shapiro 검정 결과 p-value > 0.05이면 정규성(정규분포)을 만족하는 것으로 볼 수 있다.  


In [3]:
shapiro(x1)

ShapiroResult(statistic=0.978407084941864, pvalue=0.9708739519119263)

In [4]:
shapiro(x2)

ShapiroResult(statistic=0.9537350535392761, pvalue=0.6920550465583801)

#### 등분산 가정: levene()
두 집단 데이터에 대한 등분산 가정을 확인하기 위해 levene()함수 적용

> levene(data1, data2)


2표본 t-검정은 두 집단의 평균을 비교할 때 사용하는 검정으로 1)독립표본 t-검정(서로 독립인 두 집단을 비교), 2)대응표본 t-검정(어떤 처리 전후를 비교)로 나눌 수 있다.  

이 둘은 두 집단의 등분산 여부에 따라 나눌 수 있는데 levene's test가 기각되면 이분산 가정 t-검정을 수행하고 levene's test가 기각되지 않으면 등분산 가정 t-검정을 수행해야 한다.  

t-검정에서는 중심극한정리를 사용하여 정규분포를 가정하기 때문에 표본의 수가 30개 이상이어야 한다.  
표본의 수가 30개 미만인 경우에는 정규성 검정을 통해 정규성을 입증해야 한다. (그래서 위에서 shapiro검정을 하였다.). 

만약 정규성 검정이 기각될 경우 비모수적 방법인 윌콕슨 순위합 검정을 사용해야 한다. 또 샘플사이즈가 너무 작을 경우 정규성 검정을 할 수 없다.  

주어진 x1, x2 데이터에 대해 levene's test를 한 결과 p-value > 0.05이므로 등분산을 만족하는 것을 알 수 있다.  

In [5]:
levene(x1, x2)

LeveneResult(statistic=0.09230190178073239, pvalue=0.7640023844571427)

#### 독립표본 t-검정: ttest_ind()

> ttest_ind(data1, data2, equal_var=True/False, alternative="two-sided"/"less"/"greater")

앞서 수행한 등분산 검정(levene's test 결과)가 등분산이면 True, 등분산이 아니면 False  
alternative는 검정방법에 따라 양측검정("two-sided")가 default이며, 단측검정일 경우 방향에 따라 좌측이면 "less", 우측이면 "right"를 입력한다. 지금은 차이가 있는 것인지 확인하기 위한 것이기 때문에 기본값인 양측검정을 사용하면 된다.    

참고) https://m.blog.naver.com/mykepzzang/220886418140


In [6]:
import numpy as np

In [7]:
print("one_sample 평균1)", np.mean(x1))

one_sample 평균1) 45.15384615384615


In [8]:
print("one_sample 평균2)", np.mean(x2))

one_sample 평균2) 42.25


In [9]:
print("one_sample 표준편차1)", np.std(x1))

one_sample 표준편차1) 7.6846115346105694


In [10]:
print("one_sample 표준편차2)", np.std(x2))

one_sample 표준편차2) 8.367845202519781


위 결과를 보면 x1의 평균이 x2보다 크고, x2의 표준편차가 x1보다 큰 것을 알 수 있다.  
하지만 이는 표본의 평균과 표준편차간 차이이고, 모집단의 차이를 보기 위해서는 t-test를 수행해야 한다.  


In [11]:
result = ttest_ind(x1, x2)
result

Ttest_indResult(statistic=0.8675501422139809, pvalue=0.39460232464127154)

In [12]:
print("t 검정 통계량: %.5f, p값: %.5f"%result)

t 검정 통계량: 0.86755, p값: 0.39460


t-test 결과 유의수준(p-value) 0.05 이상으로 귀무가설(H0: u1=u2) 기각 불가.   

__귀무가설을 기각할 수 없기 때문에 x1과 x2는 통계적 차이가 없다__

만약 x1과 x2의 차이가 없다는 양측검정이 아니라 x1이 x2보다 크다는 단측검정을 했다면 

In [13]:
result_g = ttest_ind(x1, x2, alternative="greater")
result_g

Ttest_indResult(statistic=0.8675501422139809, pvalue=0.19730116232063577)

In [14]:
print("t 검정 통계량: %.5f, p값: %.5f"%result_g)

t 검정 통계량: 0.86755, p값: 0.19730


t-test 결과 유의수준(p-value) 0.05 이상으로 귀무가설(H0: u1 > u2) 기각 불가.   

__귀무가설을 기각할 수 없기 때문에 x1가 x2는 통계적으로 크다고 볼 수 없다__

## 2. 대응표본 t-검정 (대응되는 두 집단에 대한 평균 차이 비교)

#### 대응표본 t-검정
두 모평균 비교에 있어서 두 모집단이 서로 독립이 아닌 경우 실험단위를 동질적인 쌍으로 묶은 다음 각 쌍을 임의로 추출하여 통계적으로 비교하는 방법

참고) https://m.blog.naver.com/kjihoon0914/221219070454

예를 들어 노인우울감 개선 프로그램 효과 측정과 같이 동일한 노인에게 프로그램을 실행하여 전/후를 비교했다면 동일한 대상의 사전, 사후 비교가 되므로 두 데이터가 __독립__ 이 아닌 __대응__ 이 된다는 차이가 있다.  

이 경우, 귀무가설은 '프로그램 수행 전과 후의 우울감에 __차이가 없다__'가 되고, 대립가설은 '프로그램 수행 전과 후의 우울감에 __차이가 있다__'가 된다.  


#### 정규성 검정: shaprio()
각 데이터셋의 정규성 확인을 위해 shapiro()함수를 적용

> shapiro(data)  


In [15]:
x1 = [70, 80, 72, 76, 76, 76, 72, 78, 82, 64, 74, 92, 74, 68, 84]
x2 = [68, 72, 62, 70, 58, 66, 68, 52, 64, 72, 74, 60, 74, 72, 74]

In [16]:
from scipy.stats import ttest_rel

In [17]:
shapiro(x1)

ShapiroResult(statistic=0.9687387943267822, pvalue=0.8389564752578735)

In [18]:
shapiro(x2)

ShapiroResult(statistic=0.900780975818634, pvalue=0.09782424569129944)

x1, x2 모두 shapiro 검정 결과 p-value > 0.05이므로 정규성을 만족하는 것을 확인할 수 있다.  

In [19]:
print("one_sample 평균1)", np.mean(x1))

one_sample 평균1) 75.86666666666666


In [20]:
print("one_sample 평균2)", np.mean(x2))

one_sample 평균2) 67.06666666666666


In [21]:
print("one_sample 표준편차1)", np.std(x1))

one_sample 표준편차1) 6.631909394904474


In [22]:
print("one_sample 표준편차2)", np.std(x2))

one_sample 표준편차2) 6.444291185917105


전후 데이터의 평균과 표준편차를 확인한 결과, 프로그램 실행 후(x2)가 평균값과 표준편차가 모두 작아진 것을 확인할 수 있다.  

In [23]:
result = ttest_rel(x1, x2, alternative="greater")
result

Ttest_relResult(statistic=3.105360487466109, pvalue=0.0038747180533270594)

In [24]:
print("t 검정 통계량: %.5f, p값: %.5f"%result)

t 검정 통계량: 3.10536, p값: 0.00387


t-test 결과 p-value <= 0.05이므로 귀무가설을 기각하고 대립가설을 수용하므로 프로그램의 효과가 있다는 결론을 내릴 수 있다.  


## 3. 카이제곱 검정 (두 범주형 변수에 대한 독립성 여부 검정)

### 3-1. 카이제곱 검정
두 범주형 변수 간의 상호관련성이 존재하는지 확인하기 위한 검정

예) 선호하는 치킨메뉴/부위(=범주형 변수)에 성별간의 차이가 있을까?

#### 개념
두 범주형 변수 사이에 어떠한 연관관계가 있는지를 검정하는 방법. 
귀무가설(H0): 두 범주형 변수는 서로 독립이다. (= 관계가 없다.). 

예) 불면증을 가지고 있는 경우와 우울증을 가지고 있는 경우가 관계가 있을까?  

참고) https://velog.io/@rsj9987/카이제곱-검정

In [25]:
import seaborn as sns

In [26]:
df = sns.load_dataset('penguins')
df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
339  Gentoo     Biscoe             NaN            NaN                NaN   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
0         3750.0    Male  
1         3800.0  Female  
2         3250.0  Female  
3            NaN     NaN  
4         3450.0  Female  
..           ...     ...  
339          NaN     NaN  
340       4850.0  Female  
341       5750.0    Male  
342       5200.0  Female  
343       5400.0    Male  

[344 rows x 7 columns]

펭귄 데이터를 이용하여 아래와 같은 귀무가설과 대립가설을 세운다.  

- 귀무가설: 서식지(island)와 펭귄의 종(species)은 관계가 없다.  
- 대립가설: 서식지(island)와 펭귄의 종(species)은 관계가 있다.  

### 3-2. 이원분할표 만들기
빈도표 작성을 위해 pandas 모듈의 crosstab()함수 사용

> crosstab(행에 위치시킬 변수, 열에 위치시킬 변수, margins=False/True)

In [27]:
import pandas as pd

In [28]:
cross_data = pd.crosstab(df.species, df.island)
cross_data

island     Biscoe  Dream  Torgersen
species                            
Adelie         44     56         52
Chinstrap       0     68          0
Gentoo        124      0          0

In [29]:
cross_data1 = pd.crosstab(df.species, df.island, margins=True)
cross_data1

island     Biscoe  Dream  Torgersen  All
species                                 
Adelie         44     56         52  152
Chinstrap       0     68          0   68
Gentoo        124      0          0  124
All           168    124         52  344

### 3-3. 카이제곱검정 통계량 구하기
chi2_contingency()함수에 이원분할표를 대입하여 카이제곱검정 실시

> chi2_contingency(이원분할표)

In [30]:
from scipy.stats import chi2_contingency

In [31]:
chi2_contingency(cross_data)

(299.55032743148195,
 1.3545738297192517e-63,
 4,
 array([[74.23255814, 54.79069767, 22.97674419],
        [33.20930233, 24.51162791, 10.27906977],
        [60.55813953, 44.69767442, 18.74418605]]))

카이제곱값은 299.550, p-value는 1.35456e-63, 자유도는 4로 나타났는데 p-value가 매우 작은 값이 나타났으므로 귀무가설(= 서식지와 종은 관계가 없다)을 기각하고 대립가설을 채택함. 

In [32]:
chi2, p, dof, expected = chi2_contingency(cross_data)

In [33]:
print("chi2 검정통계량: %.5f, 자유도: %d, p값: %.5f"%(chi2, dof, p))

chi2 검정통계량: 299.55033, 자유도: 4, p값: 0.00000


In [34]:
EX = pd.DataFrame(expected)
EX.columns = cross_data.columns
EX.index = cross_data.index
EX

island        Biscoe      Dream  Torgersen
species                                   
Adelie     74.232558  54.790698  22.976744
Chinstrap  33.209302  24.511628  10.279070
Gentoo     60.558140  44.697674  18.744186

## 실습

### [과제6-1] 제조국에 따른 가속성능(acceleration) 차이 검정
* 분석데이터: seaborn모듈 - mpg데이터
* 연구가설: 미국과 일본 자동차의 acceleration의 평균값에 차이가 있다.  
    1. 필요한 자료를 추출하여 분석할 데이터셋 구성(origin_US, origin_JAP)
    2. 각 데이터에 대한 정규성 및 등분산 검정
    3. 평균과 표본편차 확인
    4. 독립표본 t-검정 실시 및 결괏값 해석

In [35]:
df = sns.load_dataset('mpg')
df

mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    18.0          8         307.0       130.0    3504          12.0   
1    15.0          8         350.0       165.0    3693          11.5   
2    18.0          8         318.0       150.0    3436          11.0   
3    16.0          8         304.0       150.0    3433          12.0   
4    17.0          8         302.0       140.0    3449          10.5   
..    ...        ...           ...         ...     ...           ...   
393  27.0          4         140.0        86.0    2790          15.6   
394  44.0          4          97.0        52.0    2130          24.6   
395  32.0          4         135.0        84.0    2295          11.6   
396  28.0          4         120.0        79.0    2625          18.6   
397  31.0          4         119.0        82.0    2720          19.4   

     model_year  origin                       name  
0            70     usa  chevrolet chevelle malibu  
1            70     usa          buick skylark 320  
2            70     usa         plymouth satellite  
3            70     usa              amc rebel sst  
4            70     usa                ford torino  
..          ...     ...                        ...  
393          82     usa            ford mustang gl  
394          82  europe                  vw pickup  
395          82     usa              dodge rampage  
396          82     usa                ford ranger  
397          82     usa                 chevy s-10  

[398 rows x 9 columns]

##### 필요한 자료를 추출하여 분석할 데이터셋 구성(origin_US, origin_JAP)

In [36]:
# 강의안에서는 아래 방식으로 데이터 추출
# origin_USA = df.loc[df.origin == 'usa', 'acceleration']
# origin_JAP = df.loc[df.origin == 'japan', 'acceleration']

In [37]:
origin_US = df.loc[df['origin'] == 'usa']
origin_US

mpg  cylinders  displacement  horsepower  weight  acceleration  \
0    18.0          8         307.0       130.0    3504          12.0   
1    15.0          8         350.0       165.0    3693          11.5   
2    18.0          8         318.0       150.0    3436          11.0   
3    16.0          8         304.0       150.0    3433          12.0   
4    17.0          8         302.0       140.0    3449          10.5   
..    ...        ...           ...         ...     ...           ...   
392  27.0          4         151.0        90.0    2950          17.3   
393  27.0          4         140.0        86.0    2790          15.6   
395  32.0          4         135.0        84.0    2295          11.6   
396  28.0          4         120.0        79.0    2625          18.6   
397  31.0          4         119.0        82.0    2720          19.4   

     model_year origin                       name  
0            70    usa  chevrolet chevelle malibu  
1            70    usa          buick skylark 320  
2            70    usa         plymouth satellite  
3            70    usa              amc rebel sst  
4            70    usa                ford torino  
..          ...    ...                        ...  
392          82    usa           chevrolet camaro  
393          82    usa            ford mustang gl  
395          82    usa              dodge rampage  
396          82    usa                ford ranger  
397          82    usa                 chevy s-10  

[249 rows x 9 columns]

In [38]:
origin_JAP = df.loc[df['origin'] == 'japan']
origin_JAP

mpg  cylinders  displacement  horsepower  weight  acceleration  \
14   24.0          4         113.0        95.0    2372          15.0   
18   27.0          4          97.0        88.0    2130          14.5   
29   27.0          4          97.0        88.0    2130          14.5   
31   25.0          4         113.0        95.0    2228          14.0   
53   31.0          4          71.0        65.0    1773          19.0   
..    ...        ...           ...         ...     ...           ...   
382  34.0          4         108.0        70.0    2245          16.9   
383  38.0          4          91.0        67.0    1965          15.0   
384  32.0          4          91.0        67.0    1965          15.7   
385  38.0          4          91.0        67.0    1995          16.2   
390  32.0          4         144.0        96.0    2665          13.9   

     model_year origin                   name  
14           70  japan  toyota corona mark ii  
18           70  japan           datsun pl510  
29           71  japan           datsun pl510  
31           71  japan          toyota corona  
53           71  japan    toyota corolla 1200  
..          ...    ...                    ...  
382          82  japan         toyota corolla  
383          82  japan            honda civic  
384          82  japan     honda civic (auto)  
385          82  japan          datsun 310 gx  
390          82  japan       toyota celica gt  

[79 rows x 9 columns]

##### 각 데이터에 대한 정규성 및 등분산 검정

In [39]:
shapiro(origin_US['acceleration'])

ShapiroResult(statistic=0.9953622221946716, pvalue=0.6593465805053711)

In [40]:
shapiro(origin_JAP['acceleration'])

ShapiroResult(statistic=0.9857597947120667, pvalue=0.5289148092269897)

미국차와 일본차 모두 shapiro 검정 결과 p-value가 0.05보다 큰 값이 나왔으므로 acceleration 데이터가 정규성을 가진 것으로 볼 수 있다.

In [41]:
levene(origin_US['acceleration'], origin_JAP['acceleration'])

LeveneResult(statistic=8.840152557033464, pvalue=0.0031662912439459037)

p-value < 0.05이므로 등분산을 만족하지 못한다. 따라서 이후 t-test를 할 때 등분산이 아니라는 것을 equal_val=False로 입력해 주어야 한다.

##### 평균과 표본편차 확인

In [42]:
origin_US['acceleration'].mean()

15.033734939759027

In [43]:
origin_JAP['acceleration'].mean()

16.172151898734175

In [44]:
origin_US['acceleration'].std()

2.7511116715720174

In [45]:
origin_JAP['acceleration'].std()

1.9549369921554074

미국차가 평균값은 더 작게 나타나지만 표준편차는 더 크게 나타나는 것을 확인할 수 있다.  

##### 독립표본 t-검정 실시 및 결괏값 해석

In [46]:
ttest_ind(origin_US['acceleration'], origin_JAP['acceleration'], equal_var=False)

Ttest_indResult(statistic=-4.05613769340231, pvalue=7.362848024465834e-05)

t-test결과 p-value < 0.05이므로 귀무가설(미국차와 일본자동차의 acceleration의 평균값 차이는 없다)를 기각하고 차이가 있다는 대립가설을 채택한다.  

### [과제 6-2] 자동차 사고 원인 변수에 대한 차이 검정
* 분석데이터: seaborn모듈 - car_crashes 데이터
* 연구가설: 'not_distracted'와 'no_previous'변수 간의 평균차이 검증
    1. 필요한 자료를 추출하여 분석할 데이터셋 구성('not_distracted','no_previous')
    2. 각 데이터에 대한 정규성 검정
    3. 평균과 표본편차 확인
    4. 대응표본 t-검정 실시 및 결괏값 해석
    


In [47]:
df = sns.load_dataset('car_crashes')
df.head()

total  speeding  alcohol  not_distracted  no_previous  ins_premium  \
0   18.8     7.332    5.640          18.048       15.040       784.55   
1   18.1     7.421    4.525          16.290       17.014      1053.48   
2   18.6     6.510    5.208          15.624       17.856       899.47   
3   22.4     4.032    5.824          21.056       21.280       827.34   
4   12.0     4.200    3.360          10.920       10.680       878.41   

   ins_losses abbrev  
0      145.08     AL  
1      133.93     AK  
2      110.35     AZ  
3      142.39     AR  
4      165.63     CA

#### 필요한 자료를 추출하여 분석할 데이터셋 구성 ('not_distracted', 'no_previous')

In [48]:
dist = df['not_distracted']
dist.head()

0    18.048
1    16.290
2    15.624
3    21.056
4    10.920
Name: not_distracted, dtype: float64

In [49]:
prev = df['no_previous']
prev.head()

0    15.040
1    17.014
2    17.856
3    21.280
4    10.680
Name: no_previous, dtype: float64

#### 각 데이터에 대한 정규성 검정

In [50]:
shapiro(dist)

ShapiroResult(statistic=0.9856274127960205, pvalue=0.788989245891571)

In [51]:
shapiro(prev)

ShapiroResult(statistic=0.9862307906150818, pvalue=0.8144702315330505)

distracted와 previous 모두 p-value > 0.05이므로 정규성을 가진다고 볼 수 있음.

shapiro 검정은 오차항이 정규분포를 따르는지 알아보는 검정으로, 회귀분석에서 모든 독립변수에 대해서 종속변수가 정규분포를 따르는지 알아보는 방법이다.

귀무가설은 ‘H0:정규분포를 따른다’는 것으로 p-value가 0.05보다 크면 정규성을 가정하게 된다.

#### 평균과 표본편차 확인

In [52]:
dist.mean()

13.573176470588232

In [53]:
prev.mean()

14.004882352941172

In [54]:
dist.std()

4.508977084465533

In [55]:
prev.std()

3.7646718988355885

#### 대응표본 t-검정 실시 및 결괏값 해석

**주의!!!**  
대응표본 t-검정을 해야하기 때문에 ttest_rel()함수를 사용해야 함!

In [56]:
ttest_rel(dist, prev)

Ttest_relResult(statistic=-1.0201642224359342, pvalue=0.31256149310001075)

p-value가 0.05보다 크기 때문에 귀무가설(H0:'not_distracted'와 'no_previous'변수 간의 평균차이가 없다)을 채택한다.

In [57]:
ttest_rel(dist, prev, alternative='less')

Ttest_relResult(statistic=-1.0201642224359342, pvalue=0.15628074655000537)

distracted와 previous의 평균값 차이가 있다는 귀무가설로 교차분석을 수행하기 위하여 previous의 평균값이 작다는 귀무가설을 넣어 (alternative='less') 확인한 결과 p-value가 0.05보다 크게 나온다. 즉 귀무가설을 기각할 수 없기 때문에 두 변수의 평균값에는 차이가 없다는 결론을 얻을 수 있다.

### [6-3] 성별과 생존여부 간의 독립성 검정
* 분석데이터: seaborn 모듈 - titanic 데이터
* 연구가설: 'sex'와 'alive'변수 간의 평균차이 검증
    1. 데이터 결측값 확인
    2. 이원분할표 생성
    3. 교차분석 수행 후 결괏값 해석

In [58]:
df = sns.load_dataset('titanic')
df.head()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True

#### 데이터 결측값 확인

In [59]:
df.sex.isnull().sum()

0

In [60]:
df.alive.isnull().sum()

0

#### 이원분할표 생성

In [61]:
cross_data1 = pd.crosstab(df.sex, df.alive)
cross_data1

alive    no  yes
sex             
female   81  233
male    468  109

#### 교차분석 수행 후 결괏값 분석

In [62]:
chi2_contingency(cross_data1)

(260.71702016732104,
 1.1973570627755645e-58,
 1,
 array([[193.47474747, 120.52525253],
        [355.52525253, 221.47474747]]))

카이제곱검정 결과 p-value < 0.05이므로 귀무가설(성별과 생존은 관련이 없다)을 기각하고 대립가설(성별과 생존여부는 관련이 있음)을 채택함.  
카이제곱값은 260.717, p-value는 1.197e-58, 자유도는 1 
